swim/join_sender.go

```diff
@@ -334,6 +398,68 @@ func (j *joinSender) JoinCluster() ([]string, error) {
 	return nodesJoined, nil
 }
 
+// delayAttempt delays a join attempt by sleeping for an amount of time. The
+// amount of time is computed as an exponential backoff based on the number
+// of join attempts that have been made at the time of the function call;
+// the number of attempts is 0-based. It returns a time.Duration equal to the
+// amount of delay applied and an error if one occurs. If an error occurs, a
+// duration equal to 0 will be returned, signifying no delay has been applied.
+func (j *joinSender) delayAttempt(numAttempts int, randomizer func(int) int,
```

Comment:
you also use the `initial` and `max` from the `delayOpts` so if you want to keep them injectable I would suggest to pass in the complete `delayOpts` struct to keep them all together.